extensions/amp-story/0.1/bookend.js

```diff
@@ -15,22 +15,52 @@
  */
 import {ICONS} from './icons';
 import {EventType, dispatch} from './events';
+import {Services} from '../../../src/services';
 import {createElementWithAttributes, escapeHtml} from '../../../src/dom';
-import {dev} from '../../../src/log';
+import {dev, user} from '../../../src/log';
+import {dict} from './../../../src/utils/object';
+import {isObject} from '../../../src/types';
 import {scale, setStyles} from '../../../src/style';
 import {vsyncFor} from '../../../src/services';
 
 
+/**
+ * @typedef {{
+ *   shareProviders: !JsonObject|undefined,
+ *   relatedArticles: !Array<!./related-articles.RelatedArticleSet>
+ * }}
+ */
+export let BookendConfig;
+
+
+/**
+ * Maps share provider type to visible name.
+ * If the name only needs to be capitalized (e.g. `facebook` to `Facebook`) it
+ * does not need to be included here.
+ * @const {!JsonObject}
+ */
+const SHARE_PROVIDER_NAME = dict({
+  'gplus': 'Google+',
+  'linkedin': 'LinkedIn',
+  'whatsapp': 'WhatsApp',
+});
+
+
 // TODO(alanorozco): Use a precompiled template for performance
 const TEMPLATE =
-    `<h3 class="i-amp-story-bookend-heading">Share the story</h3>
-    <ul class="i-amp-story-bookend-share">
-      <li>
-        <div class="i-amp-story-bookend-share-icon">
-          ${ICONS.link}
-        </div>
-      </li>
-    </ul>`;
+    `<amp-carousel class="i-amp-story-share-carousel"
```

Comment:
Why are we using amp-carousel?  Can't this just be a div with overflow-x: auto?

Reply:
No particular reason. Removed. PTAL.